src/components/select/select-list/select-list.component.tsx

```diff
@@ -390,6 +394,10 @@ const SelectList = React.forwardRef(
             return;
           }
 
+          if (currentOption.props.disabled) {
```

Comment:
Praise: good catch adding this guard, I was initially confused by it but see it prevents a filter text selection

Reply:
@harpalsingh @ljemmo do you know if we should be excluding disabled Options from the filter? Currently they are displayed when there's a match but obviously can't be selected

![image](https://github.com/Sage/carbon/assets/44157880/0927bdf5-52a8-44dc-9afe-8c7e30b6470e)

Reply:
I think the requirement of a diabled select option isnt great imo, but based on the specs are someone can still see the option, then in theory it should be displayed since it "exists" but cant be selected. I will have a chat with @tempertemper and @ljemmo about this.